packages/@expo/server/src/middleware/rsc.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright © 2024 650 Industries.
+ * Copyright © 2024 dai-shi.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * https://github.com/dai-shi/waku/blob/f9111ed7d96c95d7e128b37e8f7ae2d80122218e/packages/waku/src/lib/middleware/rsc.ts#L1
+ */
+
+type ResolvedConfig = any;
+
+export type RenderRscArgs = {
+  config: ResolvedConfig;
+  input: string;
+  searchParams: URLSearchParams;
+  platform: string;
+  engine?: 'hermes' | null;
+  method: 'GET' | 'POST';
+  body?: ReadableStream | null;
+  contentType?: string | undefined;
+  moduleIdCallback?: ((id: string) => void) | undefined;
+};
+
+export const decodeInput = (encodedInput: string) => {
+  if (encodedInput === 'index.txt') {
+    return '';
+  }
+  if (encodedInput?.endsWith('.txt')) {
+    return encodedInput.slice(0, -'.txt'.length);
+  }
+  const err = new Error('Invalid encoded input');
+  (err as any).statusCode = 400;
+  throw err;
+};
+
+// Production / Development API Route for handling RSC. Must be applied to the RSC paths, e.g. `/_flight/[...slug]+api.tsx`
+export function getRscMiddleware(options: {
+  config: ResolvedConfig;
+  baseUrl: string;
+  rscPath: string;
+  renderRsc: (args: RenderRscArgs) => Promise<ReadableStream<any>>;
+}): {
+  GET: (req: Request) => Promise<Response>;
+  POST: (req: Request) => Promise<Response>;
+} {
+  let rscPathPrefix = options.rscPath;
+  if (rscPathPrefix !== '/' && !rscPathPrefix.endsWith('/')) {
+    rscPathPrefix += '/';
+  }
+
+  async function getOrPostAsync(req: Request): Promise<Response> {
+    const url = new URL(req.url);
+    const { method } = req;
+    if (method !== 'GET' && method !== 'POST') {
+      throw new Error(`Unsupported method '${method}'`);
+    }
+
+    const platform = url.searchParams.get('platform') ?? req.headers.get('expo-platform');
+    if (typeof platform !== 'string' || !platform) {
+      return new Response('Missing expo-platform header or platform query parameter', {
+        status: 500,
+        headers: {
+          'Content-Type': 'text/plain',
+        },
+      });
+    }
+
+    const engine = url.searchParams.get('transform.engine');
+
+    // TODO: Dev Server only?
```

Comment:
What does this todo imply?

Reply:
Updated, this might be dev-only in the future.